lib/src/back_end/code_writer.dart

```diff
@@ -119,6 +145,19 @@ class CodeWriter {
 
     _buffer.write(text);
     _column += text.length;
+
+    // If we haven't found an overflowing line yet, then this line might be one
+    // so keep track of the pieces we've encountered.
+    if (!_foundExpandLine) {
+      // TODO(perf): This [addAll()] call might be slow. If each Piece had a
```

Comment:
Which `addAll` call are you referring to?

Also, [nit] avoid using the square braces for non-doc comments

Reply:
Oops! That's a relic of an intermediate implementation that I ended up improving on. Removed.